cache/redis_cache.go

```diff
@@ -126,17 +133,37 @@ func (r *redisCache) Get(key *Key) (*CachedData, error) {
 	}
 	// since the cached results in redis are too big, we can't fetch all of them because of the memory overhead.
 	// We will create an io.reader that will fetch redis bulk by bulk to reduce the memory usage.
+	redisStreamreader := newRedisStreamReader(uint64(offset), r.client, stringKey, metadata.Length)
```

Comment:
maybe we could instantiate after the check if ttl is small?

Reply:
I don't see what it'll chance since we need the redisStreamreader whether TTL is small (it will be used to write into the tmp file) or long (it will be used to write the http response)

Reply:
oh sorry, I misread it. Forget about the comment :)